### ISSUE TITLE
Case insensitive enum deserialization feature

### ISSUE THREAD

**Issue by yissachar** [FasterXML/jackson-databind#1313]
Enums are only properly deserialized if the value is uppercase. We don't want to force all clients to have to manipulate their data into uppercase, so we would like to accept a String case-insensitively and have it be deserialized into the correct enum. 

Others have encountered this issue and there are [various solutions](http://stackoverflow.com/questions/24157817/jackson-databind-enum-case-insensitive) for achieving this effect.

I think this feature is used often enough that it warrants being added to the core, so that we could simply do something like:

```
mapper.configure(DeserializationFeature.READ_ENUMS_CASE_INSENSITIVELY, true);
```

Instead of having to search up the solution each time we encounter this issue.

**Comment by cowtowncoder** [comment]
Sounds like possibly something to support, even thought it sort of goes against grain of JSON which is case-sensitive.

Not 100% sure what the best way is, since use of `DeserializationFeature` has its benefits and drawbacks (mostly this is kind of specific setting).
One alternative might be to introduce a new flag to `@JsonFomat`, `caseSensitive`, since then it could be added to properties as well.
On the other hand, `DeserializationFeature` could already also be applied via `@JsonFormat`, so perhaps dedicated feature would be ok.

**Comment by chethanbandi** [comment]
+1

**Comment by AnaEliza** [comment]
@cowtowncoder I'd like to help you fix it. Can I have permission to create a branch?

**Comment by cowtowncoder** [comment]
@AnaEliza the usual way is to do a fork, for which you do not need extra access.

**Comment by cowtowncoder** [comment]
Fixed via #1520.
